addon/data/selector-ui.js

```diff
@@ -159,6 +159,72 @@ const ui = (function () { // eslint-disable-line no-unused-vars
+        this.chooseArchiveMode.appendChild(archiveImg);
+        this.chooseArchiveMode.appendChild(document.createTextNode("Archive Full Page"));
+        this.chooseArchiveMode.onclick = callbacks.onChooseArchiveMode;
+        this.modes.appendChild(this.chooseArchiveMode);
+        this.selectMode.appendChild(this.modes);
+        this.buttons = document.createElement("div");
+        this.cancelButton = document.createElement("button");
+        this.cancelButton.className = "pageshot-select-mode-button";
+        this.cancelButton.textContent = "Cancel";
+        this.cancelButton.style.left = "0";
+        this.cancelButton.onclick = callbacks.onCancel;
+        this.myShotsButton = document.createElement("button");
+        this.myShotsButton.className = "pageshot-select-mode-button";
+        this.myShotsButton.textContent = "Open My Shots";
+        this.myShotsButton.style.right = "0";
+        this.myShotsButton.style.backgroundColor = "#248aeb";
+        this.myShotsButton.style.color = "white";
+        this.myShotsButton.onclick = callbacks.onOpenMyShots;
+
```

Comment:
It runs in the context of the page, though somewhat sandboxed – I feel like it's worth it to avoid getting too clever.